File: hospital-backend/src/main/java/com/hospital/dao/RoleDAO.java
package com.hospital.dao;

import com.hospital.model.Role;
import java.util.List;

public interface RoleDAO {
    List<Role> getAll();
    Role getById(int id);
    Role getByName(String name);   // used by LoginServlet to resolve role string → roleId
    int  insert(Role role);
    int  update(Role role);
    int  delete(int id);
}

File: hospital/model/dao/RoleDAO.js
/**
 * DAO Interface: RoleDAO
 */
export default class RoleDAO {
  async getAll()         { throw new Error('RoleDAO.getAll() not implemented'); }
  async getById(id)      { throw new Error('RoleDAO.getById() not implemented'); }
  async getByName(name)  { throw new Error('RoleDAO.getByName() not implemented'); }
  async create(role)     { throw new Error('RoleDAO.create() not implemented'); }
  async update(id, data) { throw new Error('RoleDAO.update() not implemented'); }
  async delete(id)       { throw new Error('RoleDAO.delete() not implemented'); }
}
